### PR TITLE
nixVersions.nix_2_19: drop

### PR DESCRIPTION
**Pull request by Mic92**
We are no longer actively fixing bugs in nix 2.19. Other nix versions will follow in a bit. 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mic92** [comment]
@roberth I am trying to drop this but your hercules-ci-cnix-store depends on it.

Also nixd, cc @inclyc

**Comment by Mic92** [comment]
If you want to take maintenance over for these nix versions, let me know, but the nix team cannot support so many different nix versions.

**Comment by Mic92** [comment]
I started updating nixd for nix 2.24 here: https://github.com/nix-community/nixd/pull/613

**Comment by roberth** [comment]
I'll update the `hercules-ci-cnix-*` bindings.

**Comment by roberth** [comment]
Mini update update: I've released the updated packages to hackage. Next I'll make a backportable update by cherry-picking 3ff89d8f12b0f2b3ef4cfb9873df6c76a1d9c761, either during the weekend or on Monday.

**Comment by Mic92** [comment]
cc @inclyc for nixd release update.

**Comment by Mic92** [comment]
Last offender: https://github.com/bluskript/nix-inspect/pull/18

File: pkgs/by-name/ni/nix-inspect/package.nix [hunk 1]
```diff
@@ -42,6 +43,14 @@ in
 
     cargoHash = "sha256-/0CrHqOL4B0Rx0ZbUpW54FiisfpW6UU4uk6wctfCX5c=";
 
+    # TODO: Remove this patch when this pull request is merged and released: https://github.com/bluskript/nix-inspect/pull/18
+    patches = [
+      (fetchpatch {
```
[the remark this answers is not in the document]
Comment (msfjarvis) [reply pkgs/by-name/ni/nix-inspect/package.nix:48]: Probably needs to also be added to the `workerPackage` in the `let` block above?

[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/by-name/ni/nix-inspect/package.nix:48]: Right. Seems to not apply to the workerPackage without modification.

**Review by fricklerhandwerk — APPROVED** [review]

**Pull request merged**